src/popover/body.tsx

```diff
@@ -50,6 +49,10 @@ export default function PopoverBody({
     [onDismiss]
   );
 
+  // used to ensure that we only autoFocus the dismiss button
+  // once the popover has been repositioned
+  useEffect(() => setRendered(true), []);
```

Comment:
I guess the rendered is set to true when the popover is opened, not repositioned?

Reply:
Why is it necessary to defer the auto-focus?

Reply:
Yes, but that allows the parent PopoverContainer to complete its initial positioning logic before we try to focus the popover content. Otherwise, the PopoverBody may be outside the visible screen (that's how some of the positioning logic is done), and therefore can cause unwanted scrolling when it's focused.

Reply:
Is there some deferring logic in the library focus-lock? We can maybe update our implementation with e.g. adding a 0-timeout here: https://github.com/cloudscape-design/components/blob/main/src/internal/components/focus-lock/index.tsx#L31

Reply:
I've not been able to get to the bottom of why it worked before, but it seems like this shouldn't be the responsibility of the FocusLock component, but rather of the component that uses it, to ensure it's ready before requesting focus lock.

Reply:
went with an approach of focusing without scrolling instead, so that the exact timing is not important